### PR TITLE
Add optional embedly authentication

### PR DESCRIPTION
**Pull request by mworrell**
### Description

Optionally add the embed.ly api key to the oembed request.=

### Checklist

- [x] documentation updated
- [x] add interface to set the Embedly key in the `Auth -> External Services` part

### CODE REVIEW

**Comment by mworrell** [comment]
@ddeboer can you have a look and merge if you like?  I think all is ok, and tests do work.

File: src/models/tests/m_identity_tests.erl [hunk 1]
```diff
@@ -42,7 +42,10 @@ check_password_no_user_test() ->
     ?assertEqual({error, nouser}, m_identity:check_username_pw("mr_z", "does-not-exist", C)),
     ok.
 
-check_username_password_test() ->
+check_username_password_test_() ->
+    {timeout, 20, fun() -> check_username_password() end}.
```
Comment (ddeboer) [inline src/models/tests/m_identity_tests.erl]: Was this necessary due to some timing issue?

Comment (mworrell) [inline src/models/tests/m_identity_tests.erl]: Yes, I guess the db can be a bit slow on Travis. The default timeout is 5 seconds, which gave random timeout failures.

File: modules/mod_oembed/support/oembed_client.erl [hunk 2]
```diff
@@ -19,140 +19,78 @@
 -module(oembed_client).
 
 -author("Arjan Scherpenisse <arjan@scherpenisse.net>").
--behaviour(gen_server).
```
Comment (ddeboer) [inline modules/mod_oembed/support/oembed_client.erl]: Gen_server asynchronicity was dropped. Will this impact performance in any way or where the `handle_casts` not useful anyway?

Comment (mworrell) [inline modules/mod_oembed/support/oembed_client.erl]: It was only used to serialize requests, as only the call interface was used.
If you need something parallel or async then it is better to spawn yourself.

Besides that there was an issue where the gen_server was not restarted on failure, as it is not part of a supervisor hierarchy but just a linked process.

**Comment by mworrell** [comment]
@ddeboer Thanks for checking and merging!

**Comment by ddeboer** [comment]
Cool, thanks! This should also go to 0.x, right?

**Comment by mworrell** [comment]
I think it is useful for Driebit, as the embed is used more often now.

**Pull request merged**